code/game/machinery/bots/buttbot.dm

```diff
@@ -26,7 +26,14 @@ Here it is: Buttbot.
 	. = ..()
 	if (.)
 		return
-	fart()
+	if(can_fart())
+		speak("butt")
+		fart()
+
+/obj/machinery/bot/buttbot/proc/can_fart()
+	if(sincelastfart + 5 < world.timeofday)
+		return TRUE
+	return FALSE
```

Comment:
Why not just `return sincelastfart + 5 < world.timeofday`?

Reply:
o ya